q2doc/usage/usage.py

```diff
@@ -0,0 +1,200 @@
+import ast
+import functools
+import os
+import re
+from pathlib import Path
+
+from sphinx.util import logging
+
+import qiime2  # noqa: F401
+import pandas as pd  # noqa: F401
+from qiime2 import Artifact, Metadata  # noqa: F401
+import qiime2.sdk.usage as usage
+from q2cli.core.usage import CLIUsage
+from q2doc.usage.nodes import (
+    UsageNode,
+    UsageExampleNode,
+    UsageDataNode,
+    FactoryNode,
+)
+from qiime2.plugins import ArtifactAPIUsage
+
+from .utils import get_docname, get_ext
+from .validation import BlockValidator
+
+logger = logging.getLogger(__name__)
+
+
+def process_usage_blocks(app, doctree, _):
+    env = app.builder.env
+    os.chdir(env.srcdir)
+    drivers = env.domaindata['q2']['drivers']
+    for k, driver in drivers.items():
+        # ensure a new instance for each doc
+        use = driver() if isinstance(driver, type) else driver.__class__()
+        drivers[k] = use
```

Comment:
I believe `process_usage_blocks` will run once for every `.rst` doc. Here I am overwriting the previous documents Usage drivers.  But you could keep them in memory on the domain object, if you wanted.